public/app/features/alerting/unified/components/rules/state-history/common.ts

```diff
@@ -108,6 +109,10 @@ export const updatePanelDataWithASHFromLoki = async (panelDataProcessed: PanelDa
   const historyImplementation = getHistoryImplementation();
   const usingLokiAsImplementation = historyImplementation === StateHistoryImplementation.Loki;
 
+  if (!usingLokiAsImplementation) {
+    return panelDataProcessed;
+  }
+
```

Comment:
I would do the following and then remove the `if` from line `116` so that's easier to read with the early return.

```suggestion
if (!panelDataProcessed.alertState?.dashboardId || !panelDataProcessed.alertState?.panelId) {
    return;
}
```

Reply:
💯 done!